File: llm_generated_code/Gemini/Python/Level2/Q18.py
def longest_streak(spells):
    max_streak = 0
    current_streak = 0
    for spell in spells:
        if spell == 1:
            current_streak += 1
            max_streak = max(max_streak, current_streak)
        else:
            current_streak = 0
    return max_streak

File: llm_generated_code/Gemini/CPP/Level2/Q18.cpp
#include <iostream>
#include <vector>

using namespace std;

int longestStreak(const vector<int>& spells) {
    int n = spells.size();
    int maxStreak = 0, currentStreak = 0;

    for (int i = 0; i < n; i++) {
        if (spells[i] == 1) {
            currentStreak++;
            maxStreak = max(maxStreak, currentStreak);
        } else {
            currentStreak = 0;
        }
    }

    return maxStreak;
}

int main() {
    vector<int> spells = {1, 0, 1, 1, 0, 1, 1, 1};
    int maxStreak = longestStreak(spells);
    cout << "Longest streak of successful spells: " << maxStreak << endl;
    return 0;
}

